src/main/java/com/evansloan/collectionlog/CollectionLogManager.java

```diff
@@ -433,4 +434,31 @@ public JsonObject getUserSettingsJsonObject()
 	{
 		return jsonUtils.toJsonObject(userSettings);
 	}
+
+	public CollectionLogItem randomCollectionLog()
+	{
+		if (!isInitialized())
+		{
+			return null;
+		}
+		int totalMissing = Math.min(collectionLog.getUniqueItems() - collectionLog.getTotalObtained(), 16);
+		List<CollectionLogItem> items = new ArrayList<>(totalMissing);
+
+		for (Map.Entry<String, CollectionLogTab> entry : collectionLog.getTabs().entrySet())
+		{
+			CollectionLogTab collectionLogTab = entry.getValue();
+			for (CollectionLogPage page : collectionLogTab.getPages().values())
+			{
+				for (CollectionLogItem item : page.getItems())
```

Comment:
You can clean up this loop here by replacing it with

```java
List<CollectionLogItem> pageItems = page.applyItemFilter('missing');
items.addAll(pageItems);
```

Reply:
Will do